File: culture-travel/044-frontend/src/api/review.ts
import request from '@/utils/request'

export function addReview(data: {
  bookingId: number
  rating: number
  content?: string
  images?: string
}) {
  return request.post('/review/add', null, { params: data })
}

export function getHomestayReviews(homestayId: number, params: { current?: number; size?: number }) {
  return request.get(`/review/homestay/${homestayId}`, { params })
}

export function getMyReviews(params: { current?: number; size?: number }) {
  return request.get('/review/my', { params })
}

File: culture-travel/044-backend/src/main/java/com/xiaou/entity/Review.java
package com.xiaou.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("review")
public class Review {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long bookingId;
    private Long userId;
    private Long homestayId;
    private Integer rating;
    private String content;
    private String images;
    private String hostReply;
    private LocalDateTime replyTime;
    @TableLogic
    private Integer deleted;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
}
